### PR TITLE
Bump MSRV to 1.64 for 0.14.0

### PR DESCRIPTION
**Pull request by mtreinish**
This commit raises the MSRV for rustworkx and rustworkx-core to 1.64. Previously we had been matching the version of rust included with the latest debian stable release, which is 1.63 right now. However, some of our dependencies in rustworkx's latest release has bumped their msrv to 1.64 and instead of waiting for the next debian release it's probably better to just bump to 1.64. Installing newer versions of rust are something more users are accustomed to at this point so the need for debian compatibility isn't as important as it used to be.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ Installing on a platform without precompiled binaries
 
 If there are no precompiled binaries published for your system you'll have to
 build the package from source. However, to be able able to build the package from
-the published source package you need to have Rust >= 1.56.1 installed (and also
+the published source package you need to have Rust >= 1.63 installed (and also
```
Comment (IvanIsCoding) [inline docs/source/install.rst:28]:
```suggestion
the published source package you need to have Rust >= 1.64 installed (and also
```

File: src/dag_algo/longest_path.rs [hunk 2]
```diff
@@ -17,7 +17,7 @@ use hashbrown::HashMap;
 use pyo3::prelude::*;
 
 use petgraph::algo;
-use petgraph::graph::NodeIndex;
+use petgraph::stable_graph::NodeIndex;
```
Comment (IvanIsCoding) [inline src/dag_algo/longest_path.rs:20]: Cargo fmt complains about the alphabetical order in this one

File: README.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 [![](https://img.shields.io/github/release/Qiskit/rustworkx.svg?style=popout-square)](https://github.com/Qiskit/rustworkx/releases)
 [![](https://img.shields.io/pypi/dm/rustworkx.svg?style=popout-square)](https://pypi.org/project/rustworkx/)
 [![Coverage Status](https://coveralls.io/repos/github/Qiskit/rustworkx/badge.svg?branch=main)](https://coveralls.io/github/Qiskit/rustworkx?branch=main)
-[![Minimum rustc 1.56.1](https://img.shields.io/badge/rustc-1.56.1+-blue.svg)](https://rust-lang.github.io/rfcs/2495-min-rust-version.html)
+[![Minimum rustc 1.63](https://img.shields.io/badge/rustc-1.63+-blue.svg)](https://rust-lang.github.io/rfcs/2495-min-rust-version.html)
```
Comment (IvanIsCoding) [inline README.md:9]:
```suggestion
[![Minimum rustc 1.64](https://img.shields.io/badge/rustc-1.64+-blue.svg)](https://rust-lang.github.io/rfcs/2495-min-rust-version.html)
```

File: README.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ environment.
 
 If there are no precompiled binaries published for your system you'll have to
 build the package from source. However, to be able able to build the package
-from the published source package you need to have Rust >= 1.56.1 installed (and
+from the published source package you need to have Rust >= 1.63 installed (and
```
Comment (IvanIsCoding) [inline README.md:55]:
```suggestion
from the published source package you need to have Rust >= 1.64 installed (and
```

**Review by IvanIsCoding — APPROVED** [review]
LGTM, I think it's time to bump the MSRV

File: src/dag_algo/longest_path.rs (hunk 2 above)
Comment (mtreinish) [reply src/dag_algo/longest_path.rs]: Fixed in: https://github.com/Qiskit/rustworkx/pull/910/commits/55aeba0d4dc62b464050bbf61f1050c53d5b91a1

**Pull request merged**